### PR TITLE
preserved form data in local storage

### PR DESCRIPTION
**Pull request by Ellon-M**
## Changes
- created an object to store each form entry data.
- set each value of the form data into the local storage.
- persisted the input on page reload by retrieving the data from local storage.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -69,6 +76,19 @@ formSubmit.addEventListener('click', (e) => {
...
+    errorMsg.style.display = 'flex';
+    e.preventDefault();
+  }
+});
+
+const inputs = Object.values(formData);
+
+inputs.forEach((input) => {
+  inputPersist(input);
+});
```
Comment (franclobo) [inline script.js]: - [ ] Kindly implement all the project requirements in order to save the data in the local storage for a better user experience 👇🏽 

> You should implement the following interactions:
When the user changes the content of any input field, the data is saved to the local storage.
When the user loads the page if there is any data in the local storage the input fields are pre-filled with this data.

**HINTS**
1. We could use the `localStorage.getItem` and `localStorage.setItem` functions to save and retrieve the form data object to the local storage.
2. In order to be saved to the local storage the data object is turned into a string using `JSON.stringify`.
3. In order to be used after being retrieved from the local storage the saved string is turned back into an object using `JSON.parse`.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @Ellon-M & @prg-04, 

# Required Changes ♻️

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️ 
- Correct Github flow ✔️ 
- Descriptive PR ✔️ 
- Professional README file ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please 🙏🏽 add the information of your code partner in the `Authors` section on the README file

image

- [ ] **[OPTIONAL]** ⚠️ Kindly review the announcements in the Slack channel they informed us about a change in the [README template](https://github.com/microverseinc/readme-template) **IT IS GOING TO BE MANDATORY NEXT WEEK** ⚠️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Ellon-M** [comment]
All changes have been made as requested. Please let me know if there is anything else.

**Review by frank1738 — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Correct use of Github workflow
✅ No linter error
✅ README is okay

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**